x-pack/plugins/security_solution/public/detection_engine/rule_creation/components/history_window_start_edit/history_window_start_edit.tsx

```diff
@@ -35,24 +34,31 @@ export function HistoryWindowStartEdit({ path }: HistoryWindowStartEditProps): J
 }
 
 const HISTORY_WINDOW_START_FIELD_CONFIG: FieldConfig<HistoryWindowStart> = {
-  label: i18n.translate(
-    'xpack.securitySolution.detectionEngine.createRule.stepDefineRule.historyWindowSizeLabel',
-    {
-      defaultMessage: 'History Window Size',
-    }
-  ),
-  helpText: i18n.translate(
-    'xpack.securitySolution.detectionEngine.createRule.stepScheduleRule.historyWindowSizeHelpText',
-    {
-      defaultMessage: "New terms rules only alert if terms don't appear in historical data.",
-    }
-  ),
+  label: i18n.HISTORY_WINDOW_START_LABEL,
+  helpText: i18n.HELP_TEXT,
   validations: [
     {
-      validator: (
-        ...args: Parameters<ValidationFunc>
-      ): ReturnType<ValidationFunc<{}, ERROR_CODE>> | undefined =>
-        historyWindowStartValidationFactory(...args),
+      validator: (...args: Parameters<ValidationFunc>) => {
+        const [{ path, value }] = args;
+
+        const historyWindowSize = Number.parseInt(String(value), 10);
+
+        if (Number.isNaN(historyWindowSize)) {
+          return {
+            code: 'ERR_NOT_INT_NUMBER',
+            path,
+            message: i18n.MUST_BE_POSITIVE_INTEGER_VALIDATION_ERROR,
+          };
+        }
+
+        if (historyWindowSize <= 0) {
+          return {
+            code: 'ERR_MIN_LENGTH',
+            path,
+            message: i18n.MUST_BE_GREATER_THAN_ZERO_VALIDATION_ERROR,
+          };
+        }
+      },
```

Comment:
I'd recommend to move the validation to a separate function in a separate file for better readability.